ethergo/chain/listener/listener.go

```diff
@@ -37,11 +36,12 @@ type ContractListener interface {
 type HandleLog func(ctx context.Context, log types.Log) error
 
 type chainListener struct {
-	client   client.EVM
-	contract *fastbridge.FastBridgeRef
-	store    reldb.Service
```

Comment:
Let's make the bd generic to, made a pass on this in #2129 on there,  as I mentioned there won't be able to test until master (#2101 in particulra) gets meged into this branch, but after that #2129 should be good to go